app/src/main/java/com/github/fribourgsdp/radio/SongFragment.kt

```diff
@@ -6,42 +6,70 @@ import android.text.TextWatcher
 import android.view.View
 import android.widget.EditText
 import android.widget.TextView
+import com.github.fribourgsdp.radio.mockimplementations.MockLyricsGetter
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
+import java.lang.ClassCastException
 
 class SongFragment : MyFragment(R.layout.fragment_song) {
     private lateinit var initialLyrics : String
     private lateinit var currentLyrics : String
     private lateinit var playlist : Playlist
     private lateinit var song: Song
+    private var doSaveLyrics : Boolean = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         arguments?.let { args ->
             args.getString(PLAYLIST_DATA).let { serializedPlaylist ->
                 playlist = Json.decodeFromString(serializedPlaylist!!)
             }
+            val lyricsGetter = if (args.getString("com.github.fribourgsdp.radio.TEST") != null){
+                MockLyricsGetter
+            } else{
+                MusixmatchLyricsGetter
+            }
```

Comment:
this is not very clean because it adds not very understandable arguments and means some lines of code are never tested

Reply:
I agree, but this is the only way I found of testing this feature...

Reply:
I shall use a constant string

Reply:
to me this is still bad. I think I would prefer if you implemented a new mock class inheriting from the fragment, and in the oncreate you do what you need to do. I will let the others be the judge